### ISSUE TITLE
NetworkManager-wait-online.service times out

### ISSUE THREAD

**Issue by raboof** [NixOS/nixpkgs#296953]
### Describe the bug

When I 'nixos-rebuild switch', this takes a long time, as `NetworkManager-wait-online.service` is timing out. The network seems to work fine, though.

I set `networking.networkmanager.logLevel` to `DEBUG`, but this didn't turn up anything obvious in journalctl yet.

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.7.5, NixOS, 24.05 (Uakari), 24.05.20240316.c75037b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - channels(root): `"nixos-20.03pre194293.2436c27541b"`
 - nixpkgs: `/home/aengelen/nixpkgs`

```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by sdht0** [comment]
See #247608.

I use:

```
systemd.services.NetworkManager-wait-online = {
    serviceConfig = {
      ExecStart = [ "" "${pkgs.networkmanager}/bin/nm-online -q" ];
    };
  };
```

**Comment by raboof** [comment]
> See #247608.

Thanks for the reference! I don't think I have `networking.useNetworkd` enabled, though - unclear whether it's the same issue?

> I use:
> 
> ```
> systemd.services.NetworkManager-wait-online = {
>     serviceConfig = {
>       ExecStart = [ "" "${pkgs.networkmanager}/bin/nm-online -q" ];
>     };
>   };
> ```

haven't checked what that does, but it indeed does appear to make switching fast again :)

**Comment by sdht0** [comment]
Yeah I actually took it from https://github.com/NixOS/nixpkgs/issues/180175#issuecomment-1658731959. Found it now.

**Comment by hacscred** [comment]
This is still a problem in nixos-24.05 (specifically 6e6b3dd395c) and should be added as a release blocker for 24.11. 

People should not all be following workarounds, especially those not using networkd, since nerworkd is supposed to be the more experimental configuration.

**Comment by Enzime** [comment]
This is most likely a duplicate of https://github.com/NixOS/nixpkgs/issues/180175 which recently got fixed for Tailscale users with https://github.com/NixOS/nixpkgs/pull/344678

The fix has been backported to 24.05 in https://github.com/NixOS/nixpkgs/pull/344678 but is not available on the `nixos-24.05` channel yet

**Comment by hacscred** [comment]
Enzime, I don't run tailscale. So, the "root cause" it probably is not. It might be that the tailscale resolution describes a class of root causes, but I think I don't even have custom services depending on any network* targets. 

Nevertheless, I will install it when it becomes available.

**Comment by hacscred** [comment]
(The only place where I so something remotely related to this is (which has worked before):
``
fileSystems."/home/a" = {
    device = "nfsserver:/a";
    noCheck = true;
    fsType = "nfs4";
    options = ["x-systemd.automount" "vers=4" "x-systemd.requires=network-online.target" "noauto"];
  };
``
)

The above use case is hosting a home directory on a server and it would be nice to have such a use case covered by nixpkgs tests. I don't think it's related to this issue, but it's just doe completeness.

I tried with nixos-24.05 and I still got the same failure as described by OP.
